### DISCUSSION TITLE
How to protect routes when their JS files are available in .output/public folder?

### DISCUSSION THREAD

**Discussion by kevhof** [nuxt/nuxt#16951] [Questions]
I'm trying to implement **authorizations** on some routes (and components).

I'm aware you can create **route guards** using **middleware**, but it's not truly secure for two reasons:

1. the route javascript chunk files are available in the **public folder** `.output/public/_nuxt`;
2. middlewares run only **client-side** once the app is loaded, so it may be fiddled with by a hacker;

So how do you avoid this security leak?

**Comment by TheAlexLichter** [comment]
Only by ensuring that your API/backend does the necessariy identitiy checks. Everything in the frontend (that is not provided by an API that goes through Authentication) **can and will be visible by people who play around** ☺️

**Reply by kevhof** [reply]
I agree with what you say. But sometimes a page content cannot be downloaded using an API, because it's intricately linked to its format in `<template>` (for example a secret form to complete, where even the questions themselves should be secret). So how do you proceed in that case?
